### DISCUSSION TITLE
Parameter for restarting Docker container?

### DISCUSSION THREAD

**Discussion by ghost** [dani-garcia/vaultwarden#2271] [Q&A] [answered]
Hey everyone,

after my Vaultwarden ist running perfectly with Docker, I was wondering if there is any flag that enables the container to restart after I rebooted my VM, for example a "--restart=always" option.

Once I append this to the Docker starting command, a container is created but not started, showing nothing when using "docker ps".

Is there possibly any solution for that or a list of commands I can use with the "docker run"-command with Vaultwarden?

Thanks in advance and keep up the great work,

Oliver

**Comment by RealOrangeOne** [comment, marked as answer]
This is much more a docker question than a vaultwarden question.

`--restart=always` is exactly the way to do this. That's what it means. I'm not sure what your full `docker run` command is, but if it's not starting, then I suspect there's an error on startup. Definitely worth checking the logs. 

For managing docker containers, I definitely recommend using docker compose, as it's much easier to work with.

**Reply by ghost** [reply]
This was indeed a hint - with other containers, I tended to put the `--restart=always` at the end of the command and got an error for the specific container with this flag.

Alas, with

`docker run -d --restart=always --name vaultwarden -v /opt/vaultwarden/:/data/ -p 81:80 vaultwarden/server`

everything now works as expected.

Thanks for the hint - appreciated very well :-) !

BR,

Oliver
